contrib/globalsign/mgo/mgo.go

```diff
@@ -0,0 +1,335 @@
+func (s *Session) DB(name string) *Database {
+	return &Database{
+		Database: s.Session.DB(name),
+		cfg:      s.cfg}
+}
+
+// Collection wraps an mgo.Collection type with the context
+// and config needed to support tracing.
+type Collection struct {
+	*mgo.Collection
+	cfg mongoConfig
+}
+
+// C gets a Collection from the Mongo DB and wraps it with
+// the context and configuration from the WrapDatabase.
+func (db *Database) C(name string) *Collection {
+	return &Collection{
+		Collection: db.Database.C(name),
+		cfg:        db.cfg}
+}
+
+// Create invokes and traces Collection.Create
+func (c *Collection) Create(info *mgo.CollectionInfo) error {
+	span := newChildSpanFromContext(c.cfg)
+	err := c.Collection.Create(info)
+	span.Finish(tracer.WithError(err))
+	return err
+}
+
+// DropCollection invokes and traces Collection.DropCollection
+func (c *Collection) DropCollection() error {
+	span := newChildSpanFromContext(c.cfg)
+	err := c.Collection.DropCollection()
+	span.Finish(tracer.WithError(err))
+	return err
+}
+
+// EnsureIndexKey invokes and traces Collection.EnsureIndexKey
+func (c *Collection) EnsureIndexKey(key ...string) error {
+	span := newChildSpanFromContext(c.cfg)
+	err := c.Collection.EnsureIndexKey(key...)
+	span.Finish(tracer.WithError(err))
+	return err
+}
+
+// EnsureIndex invokes and traces Collection.EnsureIndex
+func (c *Collection) EnsureIndex(index mgo.Index) error {
+	span := newChildSpanFromContext(c.cfg)
+	err := c.Collection.EnsureIndex(index)
+	span.Finish(tracer.WithError(err))
+	return err
+}
+
+// DropIndex invokes and traces Collection.DropIndex
+func (c *Collection) DropIndex(key ...string) error {
+	span := newChildSpanFromContext(c.cfg)
+	err := c.Collection.DropIndex(key...)
+	span.Finish(tracer.WithError(err))
+	return err
+}
+
+// DropIndexName invokes and traces Collection.DropIndexName
+func (c *Collection) DropIndexName(name string) error {
+	span := newChildSpanFromContext(c.cfg)
+	err := c.Collection.DropIndexName(name)
+	span.Finish(tracer.WithError(err))
+	return err
+}
+
+// DropAllIndexes invokes and traces Collection.DropAllIndexes
+func (c *Collection) DropAllIndexes() error {
+	span := newChildSpanFromContext(c.cfg)
+	err := c.Collection.DropAllIndexes()
+	span.Finish(tracer.WithError(err))
+	return err
+}
+
+// Indexes invokes and traces Collection.Indexes
+func (c *Collection) Indexes() (indexes []mgo.Index, err error) {
+	span := newChildSpanFromContext(c.cfg)
+	indexes, err = c.Collection.Indexes()
+	span.Finish(tracer.WithError(err))
+	return indexes, err
+}
+
+// Insert invokes and traces Collectin.Insert
+func (c *Collection) Insert(docs ...interface{}) error {
+	span := newChildSpanFromContext(c.cfg)
+	err := c.Collection.Insert(docs...)
+	span.Finish(tracer.WithError(err))
+	return err
+}
+
+// Find invokes and traces Collection.Find
+func (c *Collection) Find(query interface{}) *Query {
+	return &Query{
+		Query: c.Collection.Find(query),
+		cfg:   c.cfg}
```

Comment:
Same here, and everywhere else. For consistency with the rest of the code.

Reply:
👍